src/dos/drive_virtual.cpp

```diff
@@ -427,51 +434,34 @@ bool Virtual_Drive::MakeDir(char * /*dir*/) {
 bool Virtual_Drive::TestDir(char * dir) {
 	assert(dir);
 	if (!dir[0]) return true;		//only valid dir is the empty dir
-	auto cur_file = first_file;
-	while (cur_file) {
-		if (cur_file->isdir && iequals(cur_file->name, dir)) {
-			return true;
-		}
-		cur_file = cur_file->next;
+
+	if (find_vfile_by_predicate(first_file, [dir](vfile_block_t vfile) {
+		    return vfile->isdir && iequals(vfile->name, dir);
+	    })) {
+		return true;
```

Comment:
Can you lift this predicate call + lambda definition out to a nicely named helper, like in the other code (in favour of using a more readable function call here)?